demos/motion_tracking/main_motion_tracking.py

```diff
@@ -0,0 +1,142 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+
+from threading import Thread
+from functools import partial
+from bokeh.plotting import figure, curdoc
+from bokeh.layouts import gridplot, Spacer
+from bokeh.models import LinearColorMapper, ColorBar, Title, Button
+from bokeh.models.ranges import DataRange1d
+import sys
+from multiprocessing import Pipe
+from motion_tracking_network import MotionNetwork
+
+# ==========================================================================
+# Parameters
+# ==========================================================================
+recv_pipe, send_pipe = Pipe()
+num_steps = 10
+
+# Checks whether terminate button has been clicked and allows to stop
+# updating the bokeh doc
+is_done = [False]
+# ==========================================================================
+# Set up network
+# ==========================================================================
+network = MotionNetwork(send_pipe,
+                        num_steps)
```

Comment:
Can all of this top-level code go into a
```python
def main() -> None:
    ...
```

function, which at the end also holds what is now in the `main_loop` function and is called by
```python
if __name__ == "__main__":
    main()
```

Reply:
Done

Reply:
Reverted again, bokeh does not work if we put it behind __main__, as bokeh starts the script and it is not main then.